torch/csrc/autograd/engine.cpp

```diff
@@ -31,6 +31,10 @@ namespace torch { namespace autograd {
 
 // NB: -1 indicates the CPU worker!
 static constexpr int NO_DEVICE = -2;
+
+// Every worker thread remembers, via this thread local variable, what
+// device they were supposed to be working on, in case they need to
+// remember it inside a reentrant backwards call.  See Note [Reentrant backwards]
```

Comment:
I think this is misleading. They never need to "remember" it in any special way, because it's constant after the thread is created. It's only used to detect when a reentrant call happens, so we can ensure that they will start processing their queue (but they never change queues!)